tests/ui/missing_transmute_annotations.rs

```diff
@@ -0,0 +1,17 @@
+#![warn(clippy::missing_transmute_annotations)]
+
+fn main() {
+    unsafe {
+        let x: i32 = std::mem::transmute([1u16, 2u16]);
+        //~^ ERROR: transmute used without annotations
+        let x: i32 = std::mem::transmute::<_, _>([1u16, 2u16]);
+        //~^ ERROR: transmute used without annotations
+        let x = std::mem::transmute::<_, i32>([1u16, 2u16]);
+        //~^ ERROR: transmute used without annotations
+        let x: i32 = std::mem::transmute::<[u16; 2], _>([1u16, 2u16]);
+        //~^ ERROR: transmute used without annotations
+
+        // Should not warn.
+        let x = std::mem::transmute::<[u16; 2], i32>([1u16, 2u16]);
```

Comment:
Can we also have a test for a bad `transmute` coming from an external macro?

Reply:
I added a test for macros as well.

Reply:
I meant a macro from a different crate :)
Local macros are fine to lint (like it's doing right now), but not external macros, since the user can't do much about them.

Reply:
Added as well.